### PR TITLE
docs: update engine versions in some examples

### PR DESCRIPTION
**Pull request by thaJeztah**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: docs/reference/commandline/version.md [hunk 1]
```diff
@@ -181,5 +181,5 @@ above for more information.
...
 
-{"Client":{"Platform":{"Name":"Docker Engine - Community"},"Version":"19.03.8","ApiVersion":"1.40","DefaultAPIVersion":"1.40","GitCommit":"afacb8b","GoVersion":"go1.12.17","Os":"darwin","Arch":"amd64","BuildTime":"Wed Mar 11 01:21:11 2020","Experimental":true},"Server":{"Platform":{"Name":"Docker Engine - Community"},"Components":[{"Name":"Engine","Version":"19.03.8","Details":{"ApiVersion":"1.40","Arch":"amd64","BuildTime":"Wed Mar 11 01:29:16 2020","Experimental":"true","GitCommit":"afacb8b","GoVersion":"go1.12.17","KernelVersion":"4.19.76-linuxkit","MinAPIVersion":"1.12","Os":"linux"}},{"Name":"containerd","Version":"v1.2.13","Details":{"GitCommit":"7ad184331fa3e55e52b890ea95e65ba581ae3429"}},{"Name":"runc","Version":"1.0.0-rc10","Details":{"GitCommit":"dc9208a3303feef5b3839f4323d9beb36df0a9dd"}},{"Name":"docker-init","Version":"0.18.0","Details":{"GitCommit":"fec3683"}}],"Version":"19.03.8","ApiVersion":"1.40","MinAPIVersion":"1.12","GitCommit":"afacb8b","GoVersion":"go1.12.17","Os":"linux","Arch":"amd64","KernelVersion":"4.19.76-linuxkit","Experimental":true,"BuildTime":"2020-03-11T01:29:16.000000000+00:00"}}
+{"Client":"Version":"23.0.3","ApiVersion":"1.42", ...}
```
Comment (thaJeztah) [inline docs/reference/commandline/version.md:184]: Truncated these examples, as they're really long, and the output is only to illustrate that it's outputting the information in JSON format.

File: man/src/version.md [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
+The version command prints the current version number for all independently
+versioned Docker components.
 
 # EXAMPLES
 
 ## Display Docker version information
 
-The default output:
+The default output renders all version information divided into two sections;
+the "Client" section contains information about the Docker CLI and client
```
Comment (thaJeztah) [inline man/src/version.md:9]: Mostly aligning this content more with the online reference (although there's parts not yet included in the man-pages. I left that for a later exercise).

**Comment by thaJeztah** [comment]
@dvdksn PTAL 🤗

**Review by dvdksn — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
Thanks! Let me bring this one in, and move the backport out of draft

**Pull request merged**